src/interfaces/ISignatureScheme.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8;
+
+/// @title ISignatureScheme interface
+/// @author Randamu
+/// @notice Interface for signature schemes, e.g., BN254, BLS, etc.
+interface ISignatureScheme {
+    /// @notice Returns the scheme identifier as a string, e.g., "Barreto-Naehrig Curve" or "BN254"
+    /// BN254 is an elliptic curve that belongs to the pairing-friendly curves family,
+    /// designed for efficient computation of pairing-based cryptographic
+    /// protocols (such as zk-SNARKs, zero-knowledge proofs, and other cryptographic constructions)
+    function SCHEME_ID() external view returns (string memory);
+
+    /// @notice returns the DST used in message hashing to BLS point
+    function DST() external view returns (bytes memory);
+
+    /// @notice Verifies a signature using the given signature scheme.
+    /// @param message The message that was signed. Message is a G1 point represented as bytes.
+    /// @param signature The signature to verify. Signature is a G1 point represented as bytes.
+    /// @param publicKey The public key of the signer. Public key is a G2 point represented as bytes.
+    /// @return isValid boolean which evaluates to true if the signature is valid, false otherwise.
+    function verifySignature(bytes calldata message, bytes calldata signature, bytes calldata publicKey)
```

Comment:
does an instance of signature scheme have a burned-in public key or does it take a parameter?

Reply:
I think the public key should be tied to the contract, we really want to ensure the user cannot e.g. provide both the signature and the public key themselves, or whatnot